server.py

```diff
@@ -211,40 +265,54 @@ def do_save_response():
                                 status_code=400,
                                 payload=request.args)
 
-    metadata = survey_response['metadata']
```

Comment:
This will raise a `KeyError` if metadata is missing. Is this handled in the calling code, does it need a try/except block, or do we want it to fail?